src/workerd/api/sockets.c++

```diff
@@ -66,15 +20,15 @@ jsg::Ref<Socket> connectImplNoOutputLock(
   // TODO: Validate `address` is well formed. Do I need to use `parseAddress` here or is there
   // a better way?
 
-  // Set up the connection. This is similar to HttpClientAdapter::connect.
+  // Set up the connection.
   auto headers = kj::heap<kj::HttpHeaders>(ioContext.getHeaderTable());
-  auto paf = kj::newPromiseAndFulfiller<kj::HttpClient::ConnectResponse>();
-  auto tunnel = kj::refcounted<TCPConnectResponseImpl>(kj::mv(paf.fulfiller));
-  auto promise = client->connect(address, *headers, *tunnel)
-      .attach(kj::str(address), kj::mv(headers));
-  tunnel->setPromise(kj::mv(promise));
-  kj::Promise<kj::HttpClient::ConnectResponse> connResponse = paf.promise.attach(kj::mv(tunnel));
-  return jsg::alloc<Socket>(kj::mv(connResponse));
+  auto httpClient = kj::newHttpClient(*client);
+  auto request = httpClient->connect(address, *headers);
+  // TODO(soon): If `request.status` resolves to have a statusCode < 200 || >= 300, arrange for the
+  //   the Socket to throw an appropriate error. Right now in this circumstance,
+  //   `request.connection`'s operations will throw KJ exceptions which will be exposed to the
+  //   script as internal errors.
+  return jsg::alloc<Socket>(kj::mv(request.connection));
```

Comment:
I was surprised to find that the code doesn't work (stream never receives any data). Looking into this it seems we need to attach the `status` promise:

```suggestion
  return jsg::alloc<Socket>(request.connection.attach(kj::mv(request.status)));
```

Reply:
This is weird, attaching this promise shouldn't make any difference, because `request.connection` already contains its own copy of the promise (two branches of the same `split()`). That said, this doesn't cause any real harm except extra allocations, so I'm OK with merging in this form if we cannot immediately figure out why it helps. But we need to follow up and figure it out, because it implies a bug somewhere.

Reply:
Just to confirm I double checked that I was using the right capnproto commits: my tests definitely fail without the `status` attached. 

@jasnell when you get a chance, please have a look as from chatting with Kenton this seems to be a bug in https://github.com/capnproto/capnproto/pull/1579